### ISSUE TITLE
[bug?]`import {jsx} from 'react/jsx-runtime'` would throw error in package with `type: module`

### ISSUE THREAD

**Issue by indooorsman** [evanw/esbuild#2446]
e.g.:
I bundle a lib package with latest esbuild(format: esm, target: chrome100 & `"type":"module"` in package.json), which export a react component, by default the output would be something like:
```jsx
import { jsx } from 'react/jsx-runtime';
....
```
then if some project used my lib but bundle with webpack, the error would happen:
![image](https://user-images.githubusercontent.com/4919405/183442605-b64dd16f-80f8-402d-812f-8199a24dcfd6.png)

in my opinion, if my package set type to `module` explicitly, esbuild should add `.js` to the import path: 
```jsx
import { jsx } from 'react/jsx-runtime.js';
....
```

**Comment by evanw** [comment]
This is documented here: https://esbuild.github.io/api/#jsx-import-source

> The `/jsx-runtime` and `/jsx-dev-runtime` subpaths are hard-coded by design and cannot be changed.

This is deliberate behavior.

> in my opinion, if my package set type to `module` explicitly, esbuild should add `.js` to the import path:
> 
> ```js
> import { jsx } from 'react/jsx-runtime.js';
> ```

They tried doing what you suggested, and then they reverted it because it was a bad idea: https://github.com/babel/babel/issues/12210. Closing this issue as esbuild is working as intended. Your bug report is a problem with React, not with esbuild, and this problem was fixed in React 18: https://github.com/facebook/react/issues/20235#issuecomment-1095340542.

**Comment by indooorsman** [comment]
> This is documented here: https://esbuild.github.io/api/#jsx-import-source
> 
> > The `/jsx-runtime` and `/jsx-dev-runtime` subpaths are hard-coded by design and cannot be changed.
> 
> This is deliberate behavior.
> 
> > in my opinion, if my package set type to `module` explicitly, esbuild should add `.js` to the import path:
> > ```js
> > import { jsx } from 'react/jsx-runtime.js';
> > ```
> 
> They tried doing what you suggested, and then they reverted it because it was a bad idea: [babel/babel#12210](https://github.com/babel/babel/issues/12210). Closing this issue as esbuild is working as intended. Your bug report is a problem with React, not with esbuild, and this problem was fixed in React 18: [facebook/react#20235 (comment)](https://github.com/facebook/react/issues/20235#issuecomment-1095340542).

Got it, thanks
